src/app/clusters/commodity-price-server/commodity-price-server.h

```diff
@@ -85,13 +88,22 @@ class Instance : public AttributeAccessInterface, public CommandHandlerInterface
 
     bool HasFeature(Feature aFeature) const;
 
-private:
-    Protocols::InteractionModel::Status GetMatterEpochTimeFromUnixTime(uint32_t & currentUtcTime) const;
+    // Set attribute methods
+    CHIP_ERROR SetTariffUnit(Globals::TariffUnitEnum);
+    CHIP_ERROR SetCurrency(Globals::Structs::CurrencyStruct::Type);
+    CHIP_ERROR SetCurrentPrice(const DataModel::Nullable<Structs::CommodityPriceStruct::Type>);
```

Comment:
What is the situation with the lifetime of the buffers backing the Components and Description fields?